tests/test_runner/test_waiting_branches.py

```diff
@@ -68,14 +68,17 @@ def test_run(self):
 
         runner.run()
         self.assertFalse(runner.finished)
-        self.assertSetEqual(context.done, {'start', 'blocked', 'not_blocked'})
+        self.assertSetEqual(context.done, {'start_wb', 'blocked_wb', 'not_blocked_wb'})
 
         runner.run()
         self.assertFalse(runner.finished)
-        self.assertSetEqual(context.done, {'start', 'blocked', 'not_blocked'})
+        self.assertSetEqual(context.done, {'start_wb', 'blocked_wb', 'not_blocked_wb'})
 
         context.blocked = False
 
         runner.run()
         self.assertTrue(runner.finished)
-        self.assertSetEqual(context.done, {'start', 'blocked', 'not_blocked', 'end'})
+        self.assertSetEqual(context.done, {'start_wb',
+                                           'blocked_wb',
+                                           'not_blocked_wb',
+                                           'end_wb'})
```

Comment:
All tasks were only renamed? Why?

Reply:
This is weird. That is why I pushed another branch in RHELWF-7410_add_completed_tasks_set, I'd like to show you the issue. You can test it locally and see the test results.
It seems test_waiting_branches.py will be affected by tests in test_waiting.py.
context.done will be `{'task_bb', 'task_cd', 'blocked', 'task_a', 'task_d', 'first', 'not_blocked', 'fail', 'end', 'task_x', 'task_c', 'task_b', 'task_ab', 'start', 'second'}` in `self.assertSetEqual(context.done, {'start', 'blocked', 'not_blocked'})`
But actually task like 'task_bb' are not in test_waiting_branches.py, I am not sure why it happens, maybe related to decorator, I am not sure.
Rename the task names or comment out test 'test_run' in test_waiting.py will fix the issue.